posix_test.go

```diff
@@ -401,6 +402,102 @@ func TestDeleteFile(t *testing.T) {
 	}
 }
 
+// Test posix.ReadFile()
+func TestReadFile(t *testing.T) {
+	// Create temporary directory.
+	path, err := ioutil.TempDir("", "minio-")
+	if err != nil {
+		t.Fatalf("Unable to create a temporary directory, %s", err)
+	}
+	defer removeAll(path)
+
+	// Initialize posix storage layer.
+	posix, err := newPosix(path)
+	if err != nil {
+		t.Fatalf("Unable to initialize posix, %s", err)
+	}
+
+	// Setup test environment.
+	if err = posix.MakeVol("success-vol"); err != nil {
+		t.Fatalf("Unable to create volume, %s", err)
+	}
+
+	// Create directory to make errIsNotRegular
+	if err = os.Mkdir(slashpath.Join(path, "success-vol", "object-as-dir"), 0777); err != nil {
+		t.Fatalf("Unable to create directory, %s", err)
+	}
+
+	testCases := []struct {
+		fileName    string
+		expectedErr error
+	}{
+		{"myobject", nil},
```

Comment:
4 more cases would be great 
- case with non-existant object. 
- case where fd.Seek fails, may be invalid offset. 
- case where EOF is returned by ReadFull. 
- case for errFaultyDisk.